From 3944f998fc0571bd12e16a7eafca9ff88a629407 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:28 -0400
Subject: [ethernet] igb: Fix master/slave mode for all m88 i354 PHY's

Message-id: <1409062359-13063-11-git-send-email-sassmann@redhat.com>
Patchwork-id: 89128
O-Subject: [RHEL7.1 PATCH 10/81] igb: Fix master/slave mode for all m88 i354 PHY's
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Carolyn Wyborny <carolyn.wyborny@intel.com>

This patch calls code to set the master/slave mode for all m88 gen 2
PHY's. This patch also removes the call to this function for I210 devices
only from the function that is not called by I210 devices.

Signed-off-by: Carolyn Wyborny <carolyn.wyborny@intel.com>
Tested-by: Jeff Pieper  <jeffrey.e.pieper@gmail.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit d1c17d806b6a52ff020322bec457717a91ea50a9)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/e1000_phy.c b/drivers/net/ethernet/intel/igb/e1000_phy.c
index e726675..c4c4fe3 100644
--- a/drivers/net/ethernet/intel/igb/e1000_phy.c
+++ b/drivers/net/ethernet/intel/igb/e1000_phy.c
@@ -708,11 +708,6 @@ s32 igb_copper_link_setup_m88(struct e1000_hw *hw)
   hw_dbg("Error committing the PHY changes\n");
   goto out;
  }
- if (phy->type == e1000_phy_i210) {
-  ret_val = igb_set_master_slave_mode(hw);
-  if (ret_val)
-   return ret_val;
- }
 
 out:
  return ret_val;
@@ -806,6 +801,9 @@ s32 igb_copper_link_setup_m88_gen2(struct e1000_hw *hw)
   hw_dbg("Error committing the PHY changes\n");
   return ret_val;
  }
+ ret_val = igb_set_master_slave_mode(hw);
+ if (ret_val)
+  return ret_val;
 
  return 0;
 }
-- 
1.7.1